EduVPN-multiOS/Services/NotificationsService.swift

```diff
@@ -2,123 +2,200 @@
 //  NotificationsService.swift
 //  EduVPN
 //
-//  Created by Aleksandr Poddubny on 05/06/2019.
 //  Copyright © 2020 SURFNet. All rights reserved.
 //
 
 import Foundation
 import UserNotifications
 import os.log
 
-class NotificationsService {
-    
-    struct Notification {
-        
-        let title: String
-        let body: String?
+import PromiseKit
+
+// Handles local user notifications delivered from/to the app
+
+class NotificationsService: NSObject {
+    enum NotificationCategory: String {
+        case certificateExpiry
+    }
+
+    enum CertificateExpiryNotificationAction: String {
+        case refreshCertificate
+        case ignore
+    }
+
+    // If user clicks on the 'Authorize' button or the notification itself,
+    // this block shall be called.
+    var onCertificateExpiryNotificationClicked: (() -> Void)?
+
+    static var notificationCenter: UNUserNotificationCenter {
+        UNUserNotificationCenter.current()
+    }
+
+    static var authorizationOptions: UNAuthorizationOptions = [.alert, .sound]
+
+    override init() {
+        super.init()
+        Self.notificationCenter.delegate = self
+        self.registerActionableNotifications()
     }
-    
-    func makeNotification(title: String, body: String?) -> Notification {
-        return Notification(title: title, body: body)
+
+    func registerActionableNotifications() {
+        let authorizeAction = UNNotificationAction(
+            identifier: CertificateExpiryNotificationAction.refreshCertificate.rawValue,
+            title: NSString.localizedUserNotificationString(forKey: "Renew Session", arguments: nil),
+            options: [.authenticationRequired, .foreground])
+        let ignoreAction = UNNotificationAction(
+            identifier: CertificateExpiryNotificationAction.ignore.rawValue,
+            title: NSString.localizedUserNotificationString(forKey: "Ignore", arguments: nil),
+            options: [])
+        let certificateExpiryCategory = UNNotificationCategory(
+            identifier: NotificationCategory.certificateExpiry.rawValue,
+            actions: [authorizeAction, ignoreAction],
```

Comment:
On iOS the ignore action should not exist.

Reply:
Agreed, "Ignore" does look odd in iOS. I'll remove it.